modules/core/middleware/middleware.go

```diff
@@ -0,0 +1,128 @@
+package middleware
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/cosmos/ibc-go/v3/modules/core/keeper"
+)
+
+var _ tx.Handler = ibcTxHandler{}
+
+type ibcTxHandler struct {
+	k    *keeper.Keeper
+	next tx.Handler
+}
+
+// IbcTxMiddleware implements ibc tx handling middleware
```

Comment:
This godoc should explain the functionality of the transaction middleware (dropping duplicate relays)